spec/api/email_listicle_spec.rb

```diff
@@ -0,0 +1,102 @@
+require 'rails_helper'
+
+describe "email_listicle_api", type: :api do
+  let(:v1_prefix) { "/api/v1" }
+
+  def jsonify_response(response_body)
+    JSON.parse(response_body)
+  end
+  describe "GET /all" do
+    context "when no links to return" do
+      it "returns and empty array for no links " do
+        #setup
+        allow(EmailLink).to receive(:undecided).and_return([])
```

Comment:
IMHO, we want to test that EmailLink calls `#undecided` in the system under test and returns an empty array. We don't care about the implementation of `EmailLink#undecided`, but we care that it returns an empty array. Behavior outside this expectation is an error. If we don't call `EmailLink#undecided`, the test should fail and we can update our implementation code to remove the call.

Given that `EmailLink` is an actual class in our system, and not a contrived double, I think that `expect` is more appropriate in this case.

Of course, I'm happy to hear alternative points of view. I think your reasoning for the stub is sound, but since we can use the mock as a means of keeping our code cleaner in the future I would use `expect` in this situation.

Some potentially helpful articles:

https://coderwall.com/p/e80caq/rspec-allow-vs-expect

http://stackoverflow.com/questions/28006913/rspec-allow-expect-vs-just-expect-and-return